lib/show.js

```diff
@@ -11,7 +11,7 @@ var utils = require("./utils.js");
   - `show.def(x : a): string`
 */
 function showDef(obj) {
-  return "" + obj;
+  return JSON.stringify(obj);
```

Comment:
Circularity and functions aren't the problem. For those you have to define your own `show` anyway, shouldn't rely on default (have to check function case though).